src/Sentry/TransactionTracer.cs

```diff
@@ -260,6 +281,8 @@ internal ISpan StartChild(SpanId parentSpanId, string operation)
     /// <inheritdoc />
     public void Finish()
     {
+        _idleTimer?.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
```

Comment:
There'a s race condition here where Finish could be called by the user and the background thread at the same time. 

A cheap `Interlocked.Exchange` could help here, just let the first one run through and the other thread just bails immediately